### ISSUE TITLE
Percolator seg fault

### ISSUE THREAD

**Issue by CharlesEGrant** [crux-toolkit/crux-toolkit#68]
In the process of debugging a problem with the performance test, I substituted decoy-format=peptide-reverse to become decoy-format=protein-reverse.  However, this caused Percolator to seg fault.

The output is below, and the input is attached.

Bill

../../crux percolator --output-dir tide-search --feature-file T tide-search/tide-search.target.txt
INFO: Beginning percolator.
WARNING: The output directory 'tide-search' already exists.
Existing files will not be overwritten.
INFO: CPU: Genomes-MacBook-Pro.local
INFO: Wed Dec  4 11:01:12 CET 2013
INFO: Running make-pin with 'tide-search/tide-search.target.txt' and decoy file 'tide-search/tide-search.decoy.txt'.
INFO: parsed PSM: 1000
INFO: parsed PSM: 2000
INFO: parsed PSM: 3000
INFO: parsed PSM: 4000
INFO: parsed PSM: 5000
INFO: parsed PSM: 6000
INFO: parsed PSM: 7000
INFO: parsed PSM: 8000
INFO: parsed PSM: 9000
INFO: parsed PSM: 10000
INFO: parsed PSM: 11000
INFO: parsed PSM: 12000
INFO: parsed PSM: 13000
INFO: parsed PSM: 14000
INFO: parsed PSM: 15000
INFO: parsed PSM: 16000
INFO: parsed PSM: 17000
INFO: parsed PSM: 18000
INFO: parsed PSM: 19000
INFO: parsed PSM: 20000
INFO: parsed PSM: 21000
INFO: parsed PSM: 22000
INFO: parsed PSM: 23000
INFO: parsed PSM: 24000
INFO: parsed PSM: 25000
INFO: parsed PSM: 26000
INFO: parsed PSM: 27000
INFO: parsed PSM: 28000
INFO: parsed PSM: 29000
INFO: parsed PSM: 30000
INFO: parsed PSM: 31000
INFO: parsed PSM: 32000
INFO: parsed PSM: 33000
INFO: parsed PSM: 34000
INFO: parsed PSM: 35000
INFO: parsed PSM: 36000
INFO: parsed PSM: 37000
INFO: parsed PSM: 38000
INFO: parsed PSM: 39000
INFO: parsed PSM: 40000
INFO: parsed PSM: 41000
INFO: parsed PSM: 42000
INFO: parsed PSM: 43000
INFO: parsed PSM: 44000
INFO: parsed PSM: 45000
INFO: parsed PSM: 46000
INFO: parsed PSM: 47000
INFO: parsed PSM: 48000
INFO: parsed PSM: 49000
INFO: parsed PSM: 50000
INFO: parsed PSM: 51000
INFO: parsed PSM: 52000
INFO: parsed PSM: 53000
INFO: parsed PSM: 54000
INFO: parsed PSM: 55000
INFO: parsed PSM: 56000
INFO: parsed PSM: 57000
INFO: parsed PSM: 58000
INFO: parsed PSM: 59000
INFO: parsed PSM: 60000
INFO: parsed PSM: 61000
INFO: parsed PSM: 62000
INFO: parsed PSM: 63000
INFO: parsed PSM: 64000
INFO: parsed PSM: 65000
INFO: parsed PSM: 66000
INFO: parsed PSM: 67000
INFO: parsed PSM: 68000
INFO: parsed PSM: 69000
INFO: parsed PSM: 70000
INFO: parsed PSM: 71000
INFO: parsed PSM: 1000
INFO: parsed PSM: 2000
INFO: parsed PSM: 3000
INFO: parsed PSM: 4000
INFO: parsed PSM: 5000
INFO: parsed PSM: 6000
INFO: parsed PSM: 7000
INFO: parsed PSM: 8000
INFO: parsed PSM: 9000
INFO: parsed PSM: 10000
INFO: parsed PSM: 11000
INFO: parsed PSM: 12000
INFO: parsed PSM: 13000
INFO: parsed PSM: 14000
INFO: parsed PSM: 15000
INFO: parsed PSM: 16000
INFO: parsed PSM: 17000
INFO: parsed PSM: 18000
INFO: parsed PSM: 19000
INFO: parsed PSM: 20000
INFO: parsed PSM: 21000
INFO: parsed PSM: 22000
INFO: parsed PSM: 23000
INFO: parsed PSM: 24000
INFO: parsed PSM: 25000
INFO: parsed PSM: 26000
INFO: parsed PSM: 27000
INFO: parsed PSM: 28000
INFO: parsed PSM: 29000
INFO: parsed PSM: 30000
INFO: parsed PSM: 31000
INFO: parsed PSM: 32000
INFO: parsed PSM: 33000
INFO: parsed PSM: 34000
INFO: parsed PSM: 35000
INFO: parsed PSM: 36000
INFO: parsed PSM: 37000
INFO: parsed PSM: 38000
INFO: parsed PSM: 39000
INFO: parsed PSM: 40000
INFO: parsed PSM: 41000
INFO: parsed PSM: 42000
INFO: parsed PSM: 43000
INFO: parsed PSM: 44000
INFO: parsed PSM: 45000
INFO: parsed PSM: 46000
INFO: parsed PSM: 47000
INFO: parsed PSM: 48000
INFO: parsed PSM: 49000
INFO: parsed PSM: 50000
INFO: parsed PSM: 51000
INFO: parsed PSM: 52000
INFO: parsed PSM: 53000
INFO: parsed PSM: 54000
INFO: parsed PSM: 55000
INFO: parsed PSM: 56000
INFO: parsed PSM: 57000
INFO: parsed PSM: 58000
INFO: parsed PSM: 59000
INFO: parsed PSM: 60000
INFO: parsed PSM: 61000
INFO: parsed PSM: 62000
INFO: parsed PSM: 63000
INFO: parsed PSM: 64000
INFO: parsed PSM: 65000
INFO: parsed PSM: 66000
INFO: parsed PSM: 67000
INFO: parsed PSM: 68000
INFO: parsed PSM: 69000
INFO: parsed PSM: 70000
INFO: parsed PSM: 71000
INFO: Finished make-pin.
INFO: Percolator version 2.04, Build Date Dec  3 2013 09:35:52
INFO: Copyright (c) 2006-9 University of Washington. All rights reserved.
INFO: Written by Lukas Käll (lukall@u.washington.edu) in the
INFO: Department of Genome Sciences at the University of Washington.
INFO: Issued command:
INFO: percolator -X tide-search/percolator.target.pout.xml -r tide-search/percolator.target.txt -B tide-search/percolator.decoy.txt -v 2 -P decoy_ --seed 1 -p 0.01000000 -n 0.00000000 --trainFDR 0.01000000 --testFDR 0.01000000 --maxiter 10 --train-ratio 0.60000000 --tab-out tide-search/percolator.feature.txt -s tide-search/make-pin.pin.xml
INFO: Started Wed Dec  4 11:01:36 2013
INFO: Hyperparameters fdr=0.01, Cpos=0.01, Cneg=0, maxNiter=10
INFO: WARNING: no valid local xml schema is available to validate the input.
INFO: If further errors are encountered, please reinstall Percolator.
INFO: WARNING: no valid local xml schema is available to validate the input.
INFO: If further errors are encountered, please reinstall Percolator.
INFO: enzyme=trypsin
INFO: Features:
INFO: lnrSp deltLCn deltCn Xcorr Sp IonFrac Mass PepLen Charge1 Charge2 Charge3 enzN enzC enzInt lnNumSP dM absdM 
INFO: Train/test set contains 71490 positives and 71490 negatives, size ratio=1 and pi0=1
INFO: selecting cneg by cross validation
INFO: Estimating 2051 over q=0.01 in initial direction
INFO: Reading in data and feature calculation took 27.6176 cpu seconds or 28 seconds wall time
INFO: ---Training with Cpos=0.01, Cneg selected by cross validation, fdr=0.01
INFO: Iteration 1 : After the iteration step, 2765 target PSMs with q<0.01 were estimated by cross validation
INFO: Iteration 2 : After the iteration step, 2814 target PSMs with q<0.01 were estimated by cross validation
INFO: Iteration 3 : After the iteration step, 2832 target PSMs with q<0.01 were estimated by cross validation
INFO: Iteration 4 : After the iteration step, 2830 target PSMs with q<0.01 were estimated by cross validation
INFO: Iteration 5 : After the iteration step, 2822 target PSMs with q<0.01 were estimated by cross validation
INFO: Iteration 6 : After the iteration step, 2818 target PSMs with q<0.01 were estimated by cross validation
INFO: Iteration 7 : After the iteration step, 2820 target PSMs with q<0.01 were estimated by cross validation
INFO: Iteration 8 : After the iteration step, 2820 target PSMs with q<0.01 were estimated by cross validation
INFO: Iteration 9 : After the iteration step, 2822 target PSMs with q<0.01 were estimated by cross validation
INFO: Iteration 10 :    After the iteration step, 2821 target PSMs with q<0.01 were estimated by cross validation
INFO: Obtained weights (only showing weights of first cross validation set)
INFO: # first line contains normalized weights, second line the raw weights
INFO: lnrSp deltLCn deltCn  Xcorr   Sp  IonFrac Mass    PepLen  Charge1 Charge2 Charge3 enzN    enzC    enzInt  lnNumSP dM  absdM   m0
INFO: 0.0762    0.1232  0.2300  0.1525  0.1167  0.1165  -0.0567 0.0137  0.0194  0.0035  -0.0076 0.4521  0.0699  0.0000  0.0088  0.0802  -0.0553 -1.8456
INFO: 0.1341    1.0449  2.6960  0.3445  0.0008  1.0356  -0.0001 0.0024  0.1841  0.0070  -0.0151 0.9134  0.4670  0.0000  0.0143  0.1196  -0.1206 -3.9285
INFO: After all training done, 2774 target PSMs with q<0.0100 were found when measuring on the test set
INFO: Found 2774 target PSMs scoring over 1.0000% FDR level on testset
INFO: Merging results from 3 datasets
Segmentation fault: 11
bash-3.2$ 

Reported by: wsnoble
